labs/b6.md

```diff
@@ -0,0 +1,144 @@
+
+Select this parent and press `<F9>` to kill it. Send the `SIGTERM` signal. The sleep process now has `init` as its new parent, which is PID 1. What you just did is manually orphan a process; when that happens said process is subsequently re-parented by the `init` process. 
+
+Now go through the same process again. This time, send the parent a `SIGHUP` (hangup) signal. Can you still find the sleep process? When `SIGHUP` is sent to a parent shell, the parent subsequently sends hangup signals to any child processes before terminating; all processes that receive `SIGHUP` from a parent shell will terminate -- this is one way to avoid creating orphan processes.
+
+If you are interested in learning about the different signals, run `man 7 signal`. 
+
+## Cron
+So much infrastructure in the computing world relies on scheduled processes. This is the job of the cron daemon. For example, at the OCF, we use a cron job that runs every thirty minutes to keep our servers and desktops updated via puppet. To get a feel for the cron scheduler, we're going to write a basic cron job.
+
+Open the cron editor by running `crontab -e` (if the editor of your choice isn't being launched, set the `EDITOR` environment variable), which will create a crontab for your user. Here is a sample task:
+
+```* * * * * date +"\%T" >> $HOME/timestamps.txt```
+
+Right now this runs every minute. Modify it to run every five minutes. Make sure to leave this running while you complete the remainder of the lab as you will be turning this file in.
+
+If you get stuck, visit crontab.guru!
+
+## Using systemd
+
+### What services are running right now? 
+Run `systemctl`. You’ll see a long table of every unit known to systemd.
+
+Let’s narrow it down to services for now. Run `systemctl --type=service`. Now you can see a list of all services running on your computer. Each of these services is a daemon running in the background. Do you see any familiar services running?
+
+### Controlling Services 
+Now let's use `systemd` to control a an nginx web server. Install nginx by issuing `sudo apt install nginx`. Once that is done we can tell systemd to start the service with the following: `sudo systemctl start nginx`. Run `systemctl status nginx` to ensure it is running and navigate to http://*yourvm*.decal.xcf.sh/ -- you should be greeted by the nginx default landing page.
+
+Now let's make nginx listen for connections on the nonstandard port 420. In `/etc/nginx/sites-available/default` change the following lines:
+
+```nginx
+listen 80 default_server;
+listen [::]:80 default_server;
+```
+
+to:
+```nginx
+listen 420 default_server;
+listen [::]:420 default_server;
+```
+
+Tell systemd that nginx has changed configuration and needs reloading with: `sudo systemctl reload nginx`. Now, accessing  http://*yourvm*.decal.xcf.sh/ should now give you a connection refused error and your webserver will only be accessible via http://*yourvm*.decal.xcf.sh:420/.
```

Comment:
Careful there, you're not telling systemd that your nginx configuration has changed; you're telling nginx that its own configuration has changed.

Reply:
Technically no. I'm telling systemd to run `ExecReload`, though you are right in that `ExecReload` simply runs `nginx -s reload`. Being too precise here may only serve to confuse students though. I'll go with your explanation.